docs/scaling/multi-node-setup.rst

```diff
@@ -4,162 +4,237 @@
-|                 |           |       - 10.0.1.103:4300               |
-+-----------------+-----------+---------------------------------------+
 
-.. NOTE::
+.. _manual-bootstrapping:
 
-   When adding new nodes to the cluster, you do not need to update the list of
-   hosts for existing/running nodes. The cluster will find and add new
-   nodes when they ping existing ones.
+Manual bootstrapping
+--------------------
 
-Providing a list of expected hosts is just one node discovery mechanism.
+To run a CrateDB cluster across multiple hosts, you must manually configure the
+bootstrapping process by telling nodes to:
 
-CrateDB also supports node discovery via DNS as well as discovery via API for
-clusters running on *Amazon Web Services* (AWS) or Microsoft Azure. See `the
-documentation <https://crate.io/docs/reference/configuration.html#discovery>`_
-for more information.
+  a. :ref:`Discover other nodes <discovery>`, and
+  b. :ref:`Elect a master node <master-node-election>`
 
-.. _master-node-election:
+You must also configure the :ref:`metadata gateway <metadata-gateway>` (as with
+auto-bootstrapping).
 
-Master node election
-====================
 
-In a CrateDB cluster, the master node is responsible for making changes to the
-global cluster state. The master node is elected as part of the `master node
-election`_.
+.. _discovery:
 
-CrateDB requires a *quorum* of nodes in the cluster before a master can be
-elected. This insures that multiple masters are not elected in the event of a
-network partition (which would lead to a `split-brain`_ scenario).
+Discovery
+~~~~~~~~~
 
-.. _master-node-election-4x:
+With CrateDB 4.x and above, you can configure a list of master-eligible nodes
+to `seed`_ the discovery process with the `discovery.seed_hosts`_ setting in your
+`configuration`_ file. This setting should contain one identifier per
+master-eligible node, like so:
 
-CrateDB versions 4.x and above
-------------------------------
+.. code-block:: yaml
+
+    discovery.seed_hosts:
+      - node1.example.com:4300
+      - 10.0.1.102:4300
+      - 10.0.1.103:4300
+
+Alternatively, you can configure this at startup with a command-line option:
+
+.. code-block:: console
+
+    sh$ bin/crate \
+            -Cdiscovery.seed_hosts=node1.example.com,10.0.1.102,10.0.1.103
+
+.. NOTE::
 
-CrateDB will automatically determine the ideal quorum size. However, the
-`cluster.initial_master_nodes`_ setting must be configured with a initial set
-of master-eligible nodes.
+    You must configure every node with a list of seed nodes. Each node
+    discovers the rest of the cluster via the seed nodes.
 
 .. TIP::
 
-    In a three node cluster, all nodes must declared master-eligible.
+    If you are using CrateDB 3.x or below, you can use the
+    `discovery.zen.ping.unicast.hosts`_ setting instead of
+    `discovery.seed_hosts`_.
+
+CrateDB also supports node discovery via `DNS`_ as well as discovery via API
```

Comment:
I'm wondering if this should be promoted a bit more?

Reply:
expanded this section a lot